vllm/worker/vineyard_llm_cache.py

```diff
@@ -388,40 +458,50 @@ def update_seq_kv_caches(
         for j in range(self.layer):
             self.cuda_buffer[:, j, :update_token_size].copy_(
                 kv_caches[j][:, slot_mapping // block_size, slot_mapping % block_size])
-        self.buffer.copy_(self.cuda_buffer)
+        update_buffer.copy_(self.cuda_buffer)
         end_unload.record()
         end_unload.synchronize()
         duration = start_unload.elapsed_time(end_unload) / 1000.0
         self.metrics.time_unload.append(duration)   
         self.metrics.normalized_time_unload.append(0 if update_token_size == 0 else duration/update_token_size)
-        
+
         start_time = time.perf_counter()
-        # updates into vineyard
-        updated = self.cache.update(
+
+        # async update
+        update_task = partial(self._update_kv_cache,
             prefix=update_prefix,
             tokens=update_tokens,
-            kv_cache_list=self.tensors[:update_token_size],
+            kv_cache_list=update_tensors[:update_token_size],
```

Comment:
Where is `update_tensors` initialized (i.e., filled with payload)?

Reply:
`update_tensors` is a view of `update_buffer`, payload is filled into `update_buffer`, i.e., `update_buffer.copy_(self.cuda_buffer)`